guacamole-ext/src/main/java/org/apache/guacamole/net/auth/simple/SimpleObjectPermissionSet.java

```diff
@@ -37,14 +38,77 @@
     /**
      * The set of all permissions currently granted.
      */
-    private Set<ObjectPermission> permissions = Collections.<ObjectPermission>emptySet();
+    private Set<ObjectPermission> permissions = Collections.emptySet();
 
     /**
-     * Creates a new empty SimpleObjectPermissionSet.
+     * Creates a new empty SimpleObjectPermissionSet. If you are not extending
+     * SimpleObjectPermissionSet and only need an immutable, empty
+     * ObjectPermissionSet, consider using {@link ObjectPermissionSet#EMPTY_SET}
+     * instead.
      */
     public SimpleObjectPermissionSet() {
```

Comment:
So I noticed that the other "Simple" classes seem to define (mostly) empty read-only classes that require methods to be overridden if you want them to actually contain data, but this class provides a convenient way of constructing permission sets, much in the way that `SimpleUser` used to before this PR.

I think a more consistent naming scheme might be beneficial here, as at the moment, seeing the prefix "SImple" on a class name might give a developer a false impression about what it does based on previous experience.

Reply:
I agree, though I'm not sure how that can be addressed here, at least within this PR. The various "Simple" classes have had this issue for some time, where it's not entirely clear what defining principle guides what each implementation within `org.apache.guacamole.net.auth.simple` should provide. Even the `...net.auth.` part of the package is a bit of a dinosaur.

I think actually fixing what you describe would need to be part of a larger effort to restructure guacamole-ext. Some of what has been done here is a step in a better direction (like the new `EMPTY_SET` convenience constants) while yet more "Simple" classes are more status quo.

Reply:
Yeah, I think you're right that this is not the right place to address these concerns. I guess as long as we keep moving in the direction of things being more usable and less confusing this is OK for now.